File: C++/source/Framework/Messaging/LightHouse/LightPulseUnregistrationMessage.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightPulseUnregistrationMessage.h"
#include "Framework/Messaging/LightHouse/LightPulseDispatchObjectManager.h"
#include "LightPulseDispatchTypes.h"

namespace WaveNs
{

LightPulseUnregistrationMessage::LightPulseUnregistrationMessage ()
    : WaveMessage (LightPulseDispatchObjectManager::getWaveServiceId (), LIGHT_PULSE_DISPATCH_UNREGISTER_FOR_LIGHT_PULSE)
{
}

LightPulseUnregistrationMessage::LightPulseUnregistrationMessage (const string &lightPulseName)
    : WaveMessage      (LightPulseDispatchObjectManager::getWaveServiceId (), LIGHT_PULSE_DISPATCH_UNREGISTER_FOR_LIGHT_PULSE),
      m_lightPulseName (lightPulseName)
{
}

LightPulseUnregistrationMessage::~LightPulseUnregistrationMessage ()
{
}

string LightPulseUnregistrationMessage::getLightPulseName () const
{
    return (m_lightPulseName);
}

void LightPulseUnregistrationMessage::setLightPulseName (const string &lightPulseName)
{
    m_lightPulseName = lightPulseName;
}

void LightPulseUnregistrationMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization ();

    addSerializableAttribute (new AttributeString (&m_lightPulseName, "lightPulseName"));
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightPulseUnregistrationMessage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

import com.CxWave.Wave.Framework.Messaging.Local.WaveMessage;

public class LightPulseUnregistrationMessage extends WaveMessage
{
    private String m_lightPulseName;

    public LightPulseUnregistrationMessage ()
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_UNREGISTER_FOR_LIGHT_PULSE);
    }

    public LightPulseUnregistrationMessage (final String lightPulseName)
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_UNREGISTER_FOR_LIGHT_PULSE);

        m_lightPulseName = lightPulseName;
    }

    public String getLightPulseName ()
    {
        return m_lightPulseName;
    }

    public void setLightPulseName (final String lightPulseName)
    {
        m_lightPulseName = lightPulseName;
    }
}
